### PR TITLE
Added kFactor() and kFactorPooled() to DistFunc.

### PR DESCRIPTION
**Pull request by regislebrun**
This method allows to compute the bilateral margin factor for a Normal
population with unknown mean and variance. It closes ticket #774.

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/KFactorFunctions.cxx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  mutable KernelFunction kernel_;
+}; // ConstraintFunction
+
+NumericalScalar KFactor(const NumericalScalar n,
+                        const NumericalScalar nu,
+                        const NumericalScalar p,
+                        const NumericalScalar alpha)
+{
+  if (n <= 1.0) throw InvalidArgumentException(HERE) << "Error: n must be greater than 1, here n=" << n;
+  if (nu <= 0.0) throw InvalidArgumentException(HERE) << "Error: nu must be positive, here nu=" << nu;
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/KFactorFunctions.cxx]: !(nu > 0.0)

File: python/src/DistFunc_doc.i.in [hunk 2]
```diff
@@ -49,3 +49,99 @@ Examples
...
+
+alpha : float :math:`0<\alpha<1`
+    The confidence level of the covering interval
+
+Returns
+-------
+k : float
+    The margin factor
+
+Notes:
```
Comment (jschueller) [inline python/src/DistFunc_doc.i.in]: no ":" after notes

File: python/src/DistFunc_doc.i.in [hunk 3]
```diff
@@ -49,3 +49,99 @@ Examples
 >>> import openturns as ot
 >>> ot.RandomGenerator.SetSeed(0)
 >>> r = ot.DistFunc.rNormal()"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::DistFunc::kFactorPooled
+"Exact margin factor for bilateral covering interval of pooled Normal populations
```
Comment (jschueller) [inline python/src/DistFunc_doc.i.in]: should end with a dot

File: python/src/DistFunc_doc.i.in [hunk 4]
```diff
@@ -49,3 +49,99 @@ Examples
...
+    
+Examples
+--------
+>>> import openturns as ot
+>>> k = ot.DistFunc.kFactor(5, 3, 0.95, 0.9)"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::DistFunc::kFactor
+"Exact margin factor for bilateral covering interval of a Normal population
```
Comment (jschueller) [inline python/src/DistFunc_doc.i.in]: same

**Comment by jschueller** [comment]
the tests timeout on x86

**Comment by dbarbier** [comment]
Timeouts in x86 are often caused by extended precision; in Brent::solve, I guess that a and c variables should be marked volatile, otherwise `if (a == c)` may return false even if values are the same.

**Comment by regislebrun** [comment]
@dbarbier Thanks for the trick!

**Pull request merged**